source/docs/casper/resources/tutorials/advanced/cross-contract.md

```diff
@@ -0,0 +1,681 @@
+
+Prepare the `call` contract method as described below:
+
+```bash
+
+#[no_mangle]
+pub extern "C" fn call() {
+    
+    // create a new vector - this will be the signature of the entrypoint
+    let mut vec = Vec::new();
+    vec.push(Parameter::new("new_message", CLType::String));
+    vec.push(Parameter::new("hello_world_contract", CLType::Key));
+
+    // In the named keys of the contract, add a key for the count.
+    let mut named_keys = NamedKeys::new();
+
+    // create an Entry Point Object
+    let mut entry_points = EntryPoints::new();
+
+    // add the entry point to the entry points object
+    entry_points.add_entry_point(EntryPoint::new(
+       "call_contract_2",
+       vec,
+       CLType::Unit,
+       EntryPointAccess::Public,
+       EntryPointType::Contract
+    ));
+
+    // the contract is stored in the global state
+    let (stored_contract_hash, _contract_version) = storage::new_contract(
+       entry_points,                                        // entry points
+       Some(named_keys),                                    // named keys 
+       Some("contract2_package_name".to_string()),          // package name
+       Some("contract2_access_uref".to_string())            // access uref
+    );
+
+    // to access from the account - named keys of the account
+    runtime::put_key("cross_contract_2", stored_contract_hash.into());
+}
+
+```
+
+This would be the easiest implementation of the call function. There is only one entry point which accepts the key `contract2` of type `String` and the key `hello_world_contract` of the type `Key`. There aren't any named keys which will be saved in the contracts context. The contract is then stored in the global state and placed as a named key in the accounts context.
+
+As we defined the `call_contract_2` endpoint, we should implement it as well. This endpoint will take the second contract hash as an argument and call the endpoint `update_msg` with appropriate message which will then change in the second contracts context. We will be also passing the new message as a parameter. This is presented below:
+
+```bash
+#[no_mangle]
+pub extern "C" fn call_contract_2() {
+
+    // get the contract hash from the named arguments passed to the call
+    let contract_hash: ContractHash = runtime::get_named_arg::<Key>(CONTRACT_HASH)
+    .into_hash()
+    .map(|hash| ContractHash::new(hash))
+    .unwrap();
+
+    // get the value of the second parameter to which message should be changed
+    let new_value: String = runtime::get_named_arg("new_message");
+
+    // Call Counter to increment the value.
+    let _: () = runtime::call_contract(
+        contract_hash, 
+        "update_msg", 
+        runtime_args! {
+            "message" => new_value,
+        },
+    );
+
+}
+```
+
+The logic will take the contract hash from the `hello_world_contract` field and the new message from the `new_message` field and make further contract calls.
```

Comment:
I would remove this line, it's superfluous.

Reply:
Fair point. Deleted.